helm-org.el

```diff
@@ -58,9 +58,11 @@ NOTE: This will be slow on large org buffers."
   (org-show-entry))
 
 (cl-defun helm-source-org-headings-for-files (filenames
-                                              &optional (min-depth 1) (max-depth 8))
+                                              &optional (min-depth 1) (max-depth 8) (parents nil))
   (helm-build-sync-source "Org Headings"
-    :candidates (helm-org-get-candidates filenames min-depth max-depth)
+    :candidates (helm-org-get-candidates filenames min-depth max-depth parents)
+    :candidate-transformer (if parents 'reverse nil)
+    :after-init-hook 'helm-follow-mode
```

Comment:
alphapapa <notifications@github.com> writes:

> In helm-org.el:
>
>>    (helm-build-sync-source "Org Headings"
>> -    :candidates (helm-org-get-candidates filenames min-depth max-depth)
>> +    :candidates (helm-org-get-candidates filenames min-depth max-depth parents)
>> +    :candidate-transformer (if parents 'reverse nil)
>> +    :after-init-hook 'helm-follow-mode
>
>     Did you try using :follow 1 instead of :after-init-hook ?
>    
> Yes, it does activate follow-mode, but it happens before the
> preselection, causing the point-jumping behavior.

Hmm, ok.

>
>     But anyway, I am not sure it is a good idea to enable follow-mode by default (not what I want for myself anyway).
>    
> I imagine that it's not for everyone, no. But for my use case, it
> definitely helps.

You can use C-<arrow-down/up> also to have the same behavior as
follow-mode, would it fit your needs ?

> But if you think it shouldn't be, it could be left off.

I prefer not having this by default.

> I would like to find a way to offer support for it, though. :)

If the source builder use their own classes, and then use
`helm-make-source' instead of `helm-build-sync-source', one can use
`helm-setup-user-source' in its config to enable as needed follow for
these sources.


>     Also I remember now I have created helm-after-preselection-hook
>     some days ago for this same reason, but still using :follow is
>     better IMO.
>    
> I noticed that hook, but from my limited understanding, it appears
> that it's a global hook that would apply to all Helm sources--and
> definitely not all Helm sources should have follow-mode activated.

Of course.

> Is there a way to use that hook for individual sources?

If your function is a noop for other sources, it is not a problem, e.g:

    (when (string= "my-source-name" (assq 'name (helm-get-current-source)))
       [...])

-- 
Thierry
https://emacs-helm.github.io/helm/